core/CronArchive/SegmentArchiving.php

```diff
@@ -174,7 +174,14 @@ public function getReArchiveSegmentStartDate($segmentInfo)
         }
     }
 
-    private function getCreatedTimeOfSegment($storedSegment)
+    /**
+     * Retrieve the created and last edited time as date objects from the supplied segment array
+     *
+     * @param array|null $storedSegment
+     *
+     * @return array|null
+     */
+    private function getCreatedTimeOfSegment(array $storedSegment): ?array
```

Comment:
I don't think the comment is needed, why not just update the method name.

`getCreatedAndLastEditedTimeOfSegment(array $storeSegment): ?array`

The long name method may then be a smell/sign that there should be two private methods:

`getCreatedTimeOfSegment(..)`
&
`getLastEditedTimeOfSegment(..)`

Reply:
Should the parameter passed in be typed as `?array` or default null (based on comment you added).

Reply:
> I don't think the comment is needed, why not just update the method name. 

Because I think it's nitpicking and not necessary.

>Should the parameter passed in be typed as ?array or default null (based on comment you added).

I've updated the parameter definition.

Reply:
I'll disagree it's nitpicking, but I can let go of it 😄 

Thanks for doing the other updates.